tableau_utilities/scripts/cli.py

```diff
@@ -273,39 +273,28 @@ def tableau_authentication(args):
     """ Creates the Tableau server authentication from a variety of methods for passing in credentials
 
     """
-
-    # Set the defaults from the args
-    user = args.user
-    password = args.password
-    token_name = args.token_name
-    token_secret = args.token_secret
-    site = args.site_name
-    server = args.server
-    api_version = args.api_version
-
-    if args.auth in ['args_user_pass', 'args_token']:
-        print('Using auth from the args passed in')
-    # Override the defaults with information from a settings file
-    elif args.auth == 'settings_yaml':
-        print('Using auth from the settings yaml')
+    if args.settings_path:
         with open(args.settings_path, 'r') as f:
             settings = yaml.safe_load(f)
-
-        site = settings['tableau_login']['site']
-        server = settings['tableau_login']['server']
-        token_name = settings['tableau_login']['token_name']
-        token_secret = settings['tableau_login']['token_secret']
-        api_version = settings['tableau_login']['api_version']
-        user = settings['tableau_login']['user']
-        password = settings['tableau_login']['password']
-    # Override the defaults with information from the OS
-    elif args.auth == 'os_env':
-        print('Using auth OS environment')
-        site = os.getenv("TABLEAU_SITENAME")
-        server = os.getenv("TABLEAU_SERVER_ADDRESS")
-        token_name = os.getenv("TABLEAU_PERSONAL_ACCESS_TOKEN_NAME")
-        token_secret = os.getenv("TABLEAU_PERSONAL_ACCESS_TOKEN_VALUE")
-        api_version = args.api_version
+    else:
```

Comment:
I personally fine the other code format easier to read as each section can be debugged completely independently.  But if you like it like this then it should work.

Reply:
Yeah that is actually a big reason why I wanted to go this route; so the user could have their default creds in say a yaml file or their environment variables, but they could pass custom creds via the CLI commands as well.

I hear you on the debugging thing tho. I will try to come up with a happy medium. :)
Personally, I think 'debugging' should not be a part of the source code, but should be some thing that happens in CI/Testing.

Reply:
Alright, I added some debugging for this now, while still giving the flexibility of not enforcing a single authentication method. I'm hoping this will ultimately make the CLI easier to use while also simplifying code.

Happy to continue ironing this out if you still don't like this tho! :D

Reply:
Nah, if you don't want to enforce then we don't have to.